utils/build/docker/nodejs/parametric/server.js

```diff
@@ -1,38 +1,249 @@
+
+const { trace, ROOT_CONTEXT } = require('@opentelemetry/api')
+
+const { TracerProvider } = tracer
+const tracerProvider = new TracerProvider()
+tracerProvider.register()
+
+const express = require('express');
+
+const app = express();
+app.use(express.json());
+
+
+function nanoLongToHrTime ({ high = 0, low = 0 } = {}) {
+  return [
+      high * 1e3 + Math.floor(low / 1e6),
+      (low % 1e6) * 1e3,
+  ]
+}
+
+const otelStatusCodes = {
+  'UNSET': 0,
+  'OK': 1,
+  'ERROR': 2
+}
+
+const spans = {}
+const otelSpans = {}
+
+// Endpoint /trace/span/inject_headers
+app.post('/trace/span/inject_headers', (req, res) => {
+  const request = req.body;
+  const span = spans[request.span_id]
+  const http_headersDict = {}
+  const http_headers = []
+
+  tracer.inject(span, 'http_headers', http_headersDict)
+  for (const [key, value] of Object.entries(http_headersDict)) {
+      http_headers.push([key, value])
+  }
+
+  res.json({ http_headers });
+});
+
+// Additional Endpoints
+app.post('/trace/span/start', (req, res) => {
+  const request = req.body;
+  let parent
+
+  if (request.parent_id) parent = spans[request.parent_id]
+
+  if (request.origin) {
+      const traceId = parent?.traceId
+      const parentId = parent?.parentId
+
+      parent = new SpanContext({
+          traceId,
+          parentId
+      })
+      parent.origin = request.origin
+  }
+
+  const http_headers = request.http_headers || []
+  // Node.js HTTP headers are automatically lower-cased, simulate that here.
+  const convertedHeaders = {}
+  for (const [key, value] of http_headers) {
+      convertedHeaders[key.toLowerCase()] = value
+  }
+  const extracted = tracer.extract('http_headers', convertedHeaders)
+  if (extracted !== null) parent = extracted
+
+  const span = tracer.startSpan(request.name, {
+      type: request.type,
+      resource: request.resource,
+      childOf: parent,
+      tags: {
+          service: request.service
+      }
+  })
+  spans[span.context().toSpanId()] = span
+  res.json({ span_id: span.context().toSpanId(), trace_id:span.context().toTraceId(), service:request.service, resource:request.resource,});
+});
+
+app.post('/trace/span/finish', (req, res) => {
+  const id = req.body.span_id
+  const span = spans[id]
+  span.finish()
+  delete spans[id]
+  res.json({});
+});
+
+app.post('/trace/span/flush', (req, res) => {
+  const { _tracer: { _exporter: { _writer } } } = tracer
+  _writer.flush(() => {
+    res.json({});
+  })
+});
+
+app.post('/trace/span/set_meta', (req, res) => {
+  const args = req.body;
+  const spanId = args.span_id;
+  const key = args.key;
+  const value = args.value;
+  const span = spans[spanId]
+  span.setTag(key, value)
+  res.json({});
+});
+
+app.post('/trace/span/set_metric', (req, res) => {
+  const args = req.body;
+  const spanId = args.span_id;
+  const key = args.key;
+  const value = args.value;
+  spans[spanId].setMetric(key, value);
```

Comment:
The `setMetric` function does not exist. Both tags and metrics are set with `setTag`.

```suggestion
  const span = spans[spanId];
  span.setTag(key, value);
```